src/main/java/com/xpanxion/java/springboot/da1/demo/controller/student9/MemberController9.java

```diff
@@ -0,0 +1,41 @@
+package com.xpanxion.java.springboot.da1.demo.controller.student9;
+
+import com.xpanxion.java.springboot.da1.demo.model.student9.MemberHistory9;
+import com.xpanxion.java.springboot.da1.demo.service.student9.MemberHistoryService9;
+import com.xpanxion.java.springboot.da1.demo.service.student9.MemberService9;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.format.annotation.DateTimeFormat;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.time.LocalDateTime;
+import java.util.Date;
+
+@RestController
+public class MemberController9 {
+    @Autowired
+    private MemberHistoryService9 memberHistoryService9;
+
+    @Autowired
+    private MemberService9 memberService9;
+
+    @PostMapping("student9/api/v1/member/{memberId}/checkin")
+    public MemberHistory9 addMemberCheckInHistory(@PathVariable int memberId, @RequestParam @DateTimeFormat(pattern="yyyy-MM-dd H:m:s") LocalDateTime time ){
+        var member9 = memberService9.getFindById(memberId);
```

Comment:
The general idea is to put all this controller logic into the Service layer.  Can you move all this into the service, so that there is essentially just one line of code in the controller?   Do the same for the checkout endpoint.  Thanks.

Reply:
Ok yes i can do that

Reply:
I have updated the controller and pushed the updates

Reply:
👍